test/karma/README.md

```diff
@@ -1,15 +1,57 @@
 # Karma Testing
 
-Karma can be used to test final rendered output on various browsers and operating systems. Use this process to recreate
-issues and test/fix them within browsers. Tests are then locked-in and re-validated with 
-[BrowserStack](https://www.browserstack.com/) on every Github commit against the browsers which Stencil supports
-(ie11+).
+Karma can be used to test final rendered output on various browsers and operating systems.
+Use this process to recreate issues and test/fix them within browsers.
+Tests are then locked-in and re-validated with [BrowserStack](https://www.browserstack.com/).
 
 To run any tests in this directory, minor setup is required:
 1. `cd` to `test/karma` (this directory)
 2. `npm install`
 
-### Sub-directories
+## Environments
+
+The tests in this directory can be run three different ways:
+1. Locally using Chrome
+2. Through a local tunnel to Browserstack
+3. In a continuous integration (CI) environment
+
+### Running Entirely Locally
+
+Running the tests in a local environment is the quickest way to receive feedback as a part of daily development.
+By default, locally running tests will start a headless Chrome instance and run the tests against that instance.
+For additional browsers to test against, please see the [Tunnelling into BrowserStack](#tunnelling-into-browserstack) section.
+
+### Tunnelling into Browserstack
+
+Stencil engineers can run the Karma tests by tunnelling into Browserstack.
+Doing so will allow one to target the same browsers that the project targets in CI.
+However, developers should note that tunnelling into Browserstack will count against the team's allotment of running no
+more than five parallel tests at once.
+This can delay Karma tests running in a CI environment, and affect the time it takes to land a pull request.
+
+To tunnel into Browserstack, an access key and username are required.
+These can be accessed through the [Browserstack Dashboard](https://automate.browserstack.com/dashboard/v2).
+At this time, only members of the Stencil core team are granted access keys/usernames.
+If you are a member of the Stencil team and do not have access to Browserstack, please see the team lead.
+
+Once an access key and username have been acquired, tunneling into Browserstack requires invoking the tests with the
+following environment variables:
+```
+LOCAL_BROWSERSTACK=1 BROWSERSTACK_ACCESS_KEY=[KEY] BROWSERSTACK_USERNAME=[USERNAME] npm run karma
```

Comment:
should this command be `npm run test.karma.prod` ?

Reply:
Hmm actually that's a good question. It depends on the context - I had done my building and was in the `test/karma` directory when I was running this.  I'll add to the documentation to clarify

Reply:
ahh I see, that makes sense